api/solarperformanceinsight_api/compute.py

```diff
@@ -30,22 +35,326 @@ def run_job(job_id: UUID, user: str):
         job_func(job, si)
     except Exception as err:
         logger.exception("Error for job %s", job_id)
+        try:
+            details = str(err.args[0])
+        except IndexError:
+            details = f"Raised {type(err)}"
+        msg = json.dumps({"error": {"details": details}})
         with si.start_transaction() as st:
-            st.add_job_result(
-                job_id,
-                "/",
-                "error message",
-                "application/json",
-                json.dumps({"error": {"details": err.args[0]}}),
-            )
+            st.add_job_result(job_id, "/", "error message", "application/json", msg)
             st.set_job_error(job_id)
 
 
 def lookup_job_compute_function(
     job: models.StoredJob,
 ) -> Callable[[models.StoredJob, storage.StorageInterface], None]:
+    if isinstance(job.definition.parameters.job_type, models.CalculatePerformanceJob):
+        return run_performance_job
     return dummy_func
 
 
-def dummy_func(job, storage):
+def dummy_func(job, storage):  # pragma: no cover
     raise NotImplementedError("Job computation not implemented")
+
+
+def _get_data(
+    job_id: UUID, data_id: UUID, si: storage.StorageInterface, shift: dt.timedelta
+) -> pd.DataFrame:
+    """Get the data from the database and shift forward by `shift`"""
```

Comment:
I'm not thrilled about the "*and* shift forward" part of this function. It means this function does more than one useful thing. It also makes the requirements for `process_single_modelchain` harder to understand - the data has to be shifted but you still have to supply `shift` so you can unshift the results? Why not do all the shifting within `process_single_modelchain`?

Reply:
This was to simplify `generate_weather_data` and later use of the list unmodified. I can see how just iterating weather data and shifting there is easier to follow